### PR TITLE
Add conversion table for every Kubernetes version and machine image

### PR DESCRIPTION
**Pull request by pipo02mix**
Towards https://github.com/giantswarm/giantswarm/issues/23341

This PR:

- Adds matrix with Kubernetes Version and machine image so customers can rely on defaults and we only need to define target k8s version

### Testing

- [x] Fresh install works.
- [x] Upgrade from previous version works.

### Checklist

- [x] Update changelog in `CHANGELOG.md`.
- [x] Make sure `values.yaml` is valid.

### CODE REVIEW

**Comment by ericgraf** [comment]
Maintaining a Matrix's is really hard to continually maintain. Can we not just tag on the k8s version to the image name? 

ie: 
```
nodeClasses:
  default:
    image: ubuntu-2004-kube-v{{.Values.kubernetesVersion}}
``` 

The above should work even from inside the values file AFAIK.

File: helm/cluster-openstack/ci/ci-values.yaml [hunk 1]
```diff
@@ -7,7 +7,12 @@ dnsNameservers:
 - 1.1.1.1
 - 8.8.8.8
 externalNetworkID: external-network-id
-kubernetesVersion: 1.20.9
+kubernetesVersion: v1.22.9
+kubernetesVersionImages:
```
Comment (erkanerol) [inline helm/cluster-openstack/ci/ci-values.yaml:11]: Is the customer going to set this? If not, can we hide this from the user? Why don't we just append `ubuntu-2004-kube-`?

Comment (pipo02mix) [reply helm/cluster-openstack/ci/ci-values.yaml:11]: No, it is our default value for this specific cluster OpenStack version. Customer can always override it in other levels (GitOps layers)

> Why don't we just append ubuntu-2004-kube-?

the convention can change and then it fails

**Comment by pipo02mix** [comment]
> Maintaining a Matrix's is really hard to continually maintain. Can we not just tag on the k8s version to the image name?

We had a meeting last week and we discussed it. We decide to use the matrix because we might want to change the OS. More info here -> https://github.com/giantswarm/giantswarm/issues/23341#issuecomment-1315766253

**Comment by pipo02mix** [comment]
@erkanerol and I talked about the different scenarios and why this change works along them

1) Customer wants to worry about the kubernetes version and rely on GS for the default version
It enables us to do automatic upgrades and lets us certify the version/image works (CI testing pipeline will help)

2) There is a new version of k8s that has been tested successfully
GS provides a new cluster OpenStack version with this default version and the image tested is included in the matrix

3) There is a new CVE in OS
We can change the matrix with the new OS-patched version for the selected k8s version and our automation will roll the new version

From
```
kubernetesVersionImages:
  v1.22.9: "ubuntu-2004-kube-v1.22.9"
```
To
```
kubernetesVersionImages:
  v1.22.9: "ubuntu-2005-kube-v1.22.9"
```

4) Customer wants to ping a cluster in a specific version
The customer can always use a custom value (Gitops or another method) that pings the version. We don't have a way to detect the static version but we could implement it in UI/monitoring and inform customers when this happens so they can consider upgrading the cluster

5) Rare corner case. Customer needs to use an image/k8s version not listed in our matrix under their own responsibility (we did not test it). Customers still can overwrite the matrix with their values


** This changes also allow other companies to use this chart and use their own images

**Comment by pipo02mix** [comment]
yeah let's wait till we have an agreement

**Comment by pipo02mix** [comment]
We had a meeting to discuss the direction of this and we agreed on 
Also can’t remember fully, but IIRC TLDR was somewhere along the lines of:

- Our goal is to continuously release towards the customer
- We do not want gatekeepers for cluster versions
- Upgrades should be automatable
- As limitation we still need good testing to get there

So for now we keep this on hold, and most likely it will be revisited by releng team / area kaas once needed
